tests/core/test_polydata_filters.py

```diff
@@ -58,6 +58,33 @@ def test_identical_boolean(sphere):
         sphere.boolean_intersection(sphere.copy())
 
 
+@pytest.fixture()
+def poly_circle():
+    circle = pv.Circle(resolution=30)
+    return pv.PolyData(circle.points, lines=[31, *list(range(30)), 0])
+
+
+def test_decimate_polylines(poly_circle):
+    assert poly_circle.n_points == 30
+
+    decimated = poly_circle.decimate_polylines(0.5)
+    # Allow some leeway for approximtely 50%
+    assert decimated.n_points >= 14
+    assert decimated.n_points <= 16
+
+    # low maximum error will prevent decimation.
+    # Since this is a regular shape, no decimation occurs at all with suitable choice
+    decimated = poly_circle.decimate_polylines(0.5, maximum_error=0.0001)
+    assert decimated.n_points == 30
```

Comment:
```suggestion
def test_decimate_polylines(poly_circle):
    assert poly_circle.n_points == 30
    decimated = poly_circle.decimate_polylines(0.5)
    # Allow some leeway for approximtely 50%
    assert decimated.n_points >= 14
    assert decimated.n_points <= 16

def test_decimate_polylines_maximum_error(poly_circle):
    assert poly_circle.n_points == 30
    # low maximum error will prevent decimation.
    # Since this is a regular shape, no decimation occurs at all with suitable choice
    decimated = poly_circle.decimate_polylines(0.5, maximum_error=0.0001)
    assert decimated.n_points == 30
```